### PR TITLE
Feature: Consume Discord Role Removal API in Real Dev Squad Backend

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
### Description:
This pull request aims to enhance the Real Dev Squad backend by integrating and consuming the previously developed Discord Role Removal API [Real-Dev-Squad/discord-slash-commands/pull/82] . The integration allows the backend to utilize the API's functionality to remove roles from users in Discord seamlessly.

### Key Changes:
1. **API Integration**: The Real Dev Squad backend has been updated to integrate with the Discord Role Removal API. 

2. **API Request**: The backend can now send a DELETE request to the Discord Role Removal API when triggered by the corresponding action or event. The request includes the user and role identifiers required to perform the role removal operation in Discord.

3. **Error Handling**: Error handling with try and catch block will be maintained in the function consuming this service.

4. **Logging and Monitoring**: Logging and monitoring have been enhanced within the backend to capture relevant information and track the integration with the Discord Role Removal API. This allows for easier debugging, performance monitoring, and error tracking.

Testing:
Comprehensive testing has been performed to ensure the correctness and reliability of the integration. The tests cover various use cases, including successful role removal, error scenarios, and edge cases. Mocking techniques have been utilized to simulate API responses and validate the backend's behavior accurately.

Test Coverage: 
image

Please review this pull request and provide your feedback, suggestions, and any concerns you may have regarding the integration of the Discord Role Removal API into the Real Dev Squad backend.

### CODE REVIEW

File: services/discordService.js [hunk 1]
```diff
@@ -64,8 +64,20 @@ const addRoleToUser = async (userid, roleid) => {
   return response;
 };
 
+const removeRoleFromUser = async (roleId, discordId) => {
+  const authToken = await generateAuthTokenForCloudflare();
+  const data = await fetch(`${DISCORD_BASE_URL}/roles`, {
+    method: "DELETE",
+    headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
+    body: JSON.stringify({ userid: discordId, roleid: roleId }),
+  });
```
Comment (heyrandhir) [inline services/discordService.js:73]: can we add try catch block for both of these operation,

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: services/discordService.js (hunk 1 above)
Comment (RitikJaiswal75) [reply services/discordService.js:73]: error handling will be done in consuming function

**Review by heyrandhir — APPROVED** [review]
LGTM

**Comment by RitikJaiswal75** [comment]
CLosing this PE due to improper merge conflicts and resolution
